lib/omnibus/software.rb

```diff
@@ -280,6 +280,8 @@ def source(val = NULL)
             "be a kind of `Hash', but was `#{val.class.inspect}'")
         end
 
+        val = canonicalize_source(val)
```

Comment:
@sethvargo I put a `dup` in here: https://github.com/chef/omnibus/blob/master/lib/omnibus/software.rb#L1101

Sorry to respond after merge, I didn't see your comment because it appeared way up above the fold. Does this resolve what you were after? If not, I'll happily make a new PR!